freeipa/src/main/java/com/sequenceiq/freeipa/service/image/ImageService.java

```diff
@@ -91,24 +91,40 @@ public ImageEntity changeImage(Stack stack, ImageSettingsRequest imageRequest) {
         return imageRepository.save(imageEntity);
     }
 
-    public Pair<ImageWrapper, String> fetchImageWrapperAndName(Stack stack, ImageSettingsRequest imageRequest) {
+    public Pair<ImageWrapper, String> fetchImageWrapperAndName(Stack stack, FreeIpaImageFilterSettings imageFilterSettings) {
         String region = stack.getRegion();
         String platformString = platformStringTransformer.getPlatformString(stack);
-        ImageWrapper imageWrapper = getImage(imageRequest, region, platformString);
+        imageFilterSettings.setRegion(region);
+        imageFilterSettings.setPlatform(platformString);
```

Comment:
same here as before, it's usually better to handle method params as final/immutable objects